ClemBot.Bot/bot/cogs/trivia_cog.py

```diff
@@ -0,0 +1,566 @@
+
+        return
+
+    @trivia.command(aliases=['help'])
+    @ext.long_help(
+        "Lists the categories, difficulty, and type of questions. Useful for finding the index of categories!")
+    @ext.short_help(
+        "Use this to find the category you want!")
+    @commands.cooldown(1, 30, commands.BucketType.user)
+    @ext.example("!trivia help")
+    async def list_help(self, ctx):  #Overengineered this slightly. If the categories/difficulty/whatever else changes it will be a short fix
+
+        embed_list = []
+        final_page = []
+
+        category_generator = helper_fixer(CATEGORYLIST)
+        for x in category_generator:
+            embed_list.append(x)
+
+        sizeofcategory = len(embed_list)
+        x = 0
+
+        while x < sizeofcategory:
+            category_embed = discord.Embed(title="Category List:", color=Colors.ClemsonOrange)
+            category_embed.add_field(name="Index:", value=embed_list[x])
+
+            final_page.append(category_embed)
+            x += 1
+
+        difficulty_generator = helper_fixer(DIFFICULTY)
+
+        for i in difficulty_generator:
+            embed_list.append(i)
+
+        new_insertion = len(embed_list)
+
+        while sizeofcategory < new_insertion:
+            difficulty_embed = discord.Embed(title="Difficulty List:", color=Colors.ClemsonOrange)
+            difficulty_embed.add_field(name="Index:", value=embed_list[sizeofcategory])
+
+            final_page.append(difficulty_embed)
+            sizeofcategory += 1
+
+        question_generator = helper_fixer(QUESTIONTYPE)
+
+        for i in question_generator:
+            embed_list.append(i)
+
+        last_size = len(embed_list)
+        while new_insertion < last_size:
+            type_embed = discord.Embed(title="Question Type:", color=Colors.ClemsonOrange)
+            type_embed.add_field(name="Index:", value=embed_list[new_insertion])
+
+            final_page.append(type_embed)
+            new_insertion += 1
+
+        await self.bot.messenger.publish(Events.on_set_pageable_embed,
+                                         pages=final_page,
+                                         author=ctx.author,
+                                         channel=ctx.channel,
+                                         timeout=60, )
+
+        return
+
+    async def url_builder(self, function_list, input_length):
+
+        max_index = input_length - 1
+        url = URL_BUILDER + str(function_list[0])
+        x = 1
+
+
+        while x <= max_index:
+            if function_list[x]:
+                match x:
+                    case 1:
+                        url += ("&category=" + str(function_list[1]))
+                    case 2:
+                        url += ("&difficulty=" + function_list[2])
+                    case 3:
+                        url += ("&type=" + function_list[max_index])
+            x += 1
+
+        return url
+
+    async def matching_function(self, case, *input: str):
```

Comment:
As stated above this entire method will need to become custom d.py converters to remove arg parsing from the cog